lightning-invoice/src/utils.rs

```diff
@@ -300,13 +307,14 @@ where
 		InvoiceDescription::Direct(
 			&Description::new(description).map_err(SignOrCreationError::CreationError)?,
 		),
-		duration_since_epoch,
+		duration_since_epoch, invoice_expiry_delta_secs
 	)
 }
 
 fn _create_invoice_from_channelmanager_and_duration_since_epoch<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>(
 	channelmanager: &ChannelManager<Signer, M, T, K, F, L>, keys_manager: K, network: Currency,
 	amt_msat: Option<u64>, description: InvoiceDescription, duration_since_epoch: Duration,
+	invoice_expiry_delta_secs: u32
```

Comment:
We'll need to pass this to `create_inbound_payment` below instead of `DEFAULT_EXPIRY_TIME`. Otherwise, `ChannelManager` will not honor the expiry.

Reply:
Yes, will split them up :)

Reply:
Ah, I was actually wrong about this. We wouldn't need `Time` to test this because the expiry is checked against the block header time.

https://github.com/lightningdevkit/rust-lightning/blob/65920818db58880f6576fd50c3ea5df273912978/lightning/src/ln/channelmanager.rs#L5399

Reply:
Oh, interesting. So it should be pretty straightforward to test.

Reply:
Yeah, though, would probably need to modify `functional_test_utils` to set the time in a more realistic manner. Currently, it just uses the block height.